# Learning to code in isolation

### WEEK 1

1. Introduction to Unix — portal to data processing 
- At least know what is possible 
- Bash shell in Windows 
- Shortcuts and environment setup 
HW: Submit CHPC application 

2. Basics I: https://docs.google.com/presentation/d/1B8kvetTDwUe-d7hZuV2NufVOMPM7MCxh_MyP-n9yDZo/edit
HW: review 

3. Basics II: https://docs.google.com/presentation/d/1YSXYqCSHUZGRVr00oTttv_v1u83ccPLpF5_TMtW0iRI/edit
HW
	- download GTF file
	- search for where gene X is 
	- Create truncated GTF file with just Chr IV genes
	- Create truncated FASTQ file with just the first 100 reads
	


### WEEK 2

4. Anaconda, Bioconda, and SRA  
HW: Download samtools, STAR, and XPRESSpipe to environment

5. Using tools and their documentation; GitHub
	- Samtools 
	- STAR 
	- XPRESSpipe 
HW: 
	- Download BAM file 
	- Read BAM file docs 
	- Make Github Account

6. SLURM environment  and file locations; SLURM jobs
HW: Download FASTQ file and align with STAR



### WEEK 3 

7. Programming in R 
	- Googling 	
	- Data types and variables 
	- Combining strings 
	- Libraries 
	- Functions 
	- Test cases 
HW: Download R Studio, write hello world script and test on command line 

8. Programming in R 
	- Conditions 
	- Loops 
	- While 
HW: Write script and test 

9. Programming in R 
	- Dataframes 
	- S4 containers
HW: Manipulating dataframes 



### WEEK 4

10. Introduction to Data Analysis in R 
HW: Download ggplot2, DESeq2 

11. Using stats tools for data processing 
HW: DESeq2 docs and run analysis from counts table  

12. Plotting with ggplot2 
HW: Make scatter plot with count data to see if technical replicates look okay



### WEEK 5 

13. Proteomics examples
- Volcano plots, heat maps 

14. Metabolomics examples
— Alex Bott

15. Resources and continued learning 
